lib/debug.nix

```diff
@@ -14,9 +14,25 @@
 */
 { lib }:
 let
-  inherit (builtins) trace isAttrs isList isInt
-          head substring attrNames;
-  inherit (lib) id elem isFunction;
+  inherit (lib)
+    isInt
+    attrNames
+    isList
+    isAttrs
+    substring
+    addErrorContext
+    attrValues
+    concatLists
+    concatStringsSep
+    const
+    elem
+    generators
+    head
+    id
+    isDerivation
+    isFunction
+    mapAttrs
+    trace;
```

Comment:
Doesn’t importing from `lib` instead of `builtins` introduce another indirection? Sounds to me like we can get another small speedup by being explicit about the builtins; otoh that means we need to be careful to not import stuff that’s not supported in old nix versions. 
Maybe worth investigating whether there’s a small evaluation gain there.

Reply:
I had the same thought, but `lib`s top-level attribute set is only evaluated once, so the overhead of this indirection amounts to a single lookup. You really need the repeated evaluation of certain expressions to make this optimization effective.

I'd be in favor of adding more builtins to `lib` for consistency and convenience. I've only done it for ones I got wrong writing the explicit inherits, which, although admittedly a bit arbitrarily, keeps the scope of this PR in check.

Reply:
Let’s do it in a followup then.